CHANGELOG.md

```diff
@@ -7,6 +7,27 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Deprecations
+
+* `jj checkout` is now deprecated; use `jj new` instead.
+
+  **Rationale**: `jj checkout` is functionally equivalent to `jj new`, except it
+  can only take a single argument; `jj new` can take more than one argument,
+  resulting in a merge commit with multiple parents. They are always equivalent
+  in the case of a single argument, hence `jj new` is a strict superset of
+  functionality.
+  
+  More importantly, the verb `checkout` is a relic of a bygone era &mdash; the
+  days of RCS file locking, before 3-way merge &mdash; and is not a good,
+  fitting name for the functionality it provides. We encourage all users to
+  adopt `jj new` instead; it is shorter, more general, and easier to remember.
+
+  `jj checkout` will no longer be shown as part of `jj help`, but will still
+  function for now, emitting a warning about its deprecation.
+
+  **Deadline**: `jj checkout` will be deleted and become a **hard error in
+  version 0.18.0**.
```

Comment:
Nit: Remove the deadline and move a part of the rationale to the commit message. I'd say the explanation of the checkout verb should definitely be in the commit message.

Reply:
Not sure how to word this in such a short space or if it's needed; will think on it...

Reply:
> Not sure how to word this in such a short space

Who said that that the commit message should be short? I'd like to see an TLDR and then you can an even more extensible explananation behind for all I care.

Reply:
Oh, for some reason I thought you said *warning* message, not commit message.